R/te_datastore.R

```diff
@@ -170,3 +170,28 @@ setMethod(
     object
   }
 )
+
+
+#' @rdname read_expanded_data
+setMethod(
+  f = "read_expanded_data",
+  signature = "te_datastore_csv",
+  definition = function(object, period) {
+    checkmate::assert_integerish(period, null.ok = TRUE, any.missing = FALSE, lower = 0)
+    all_files <- object@files
+    if (is.null(period)) files <- all_files
+    else {
+      files <- NULL
+      for (p in period) {
+        files <- append(
+          files,
+          grep(x = all_files,
+               pattern = paste0("trial_", p, ".csv"),
+               value = TRUE)
+        )
+      }
+    }
```

Comment:
I wonder if something like this would work. 
```suggestion
    files <- if (is.null(period)) {
      all_files
    } else {
      grep(x = all_files, pattern = paste0("trial_", period, ".csv", collapse = "|"), value = TRUE)
    }
```
Here we make a regex with all of the given periods.